README.md
# Brightness Controller

This is version 3 of Brightness Controller. It supports an arbitrary number of displays!

![](img/BrightnessController.gif)

## Features

The following features are implemented:

1. Brightness Control
1. Saving color profile
1. Loading color profile


## Dependencies
There are several dependencies:

1. pyqt5
2. xrandr support in your system
3. python3

## Bugs

Please test v3. Reporting bugs is appreciated.




changelog.md
# Change Log

2018-09-12: Ported to Python 3 and PyQt5

2017-05-24: Added support for arbitary number of displays. Removed reverse control since it becomes unnecessary.


src/ui/help.py
# -*- coding: utf-8 -*-

from PyQt5 import QtCore, QtWidgets

class Ui_Form(object):
    def setupUi(self, Form):
        Form.setObjectName("Form")
        Form.resize(400, 300)
        Form.setMinimumSize(QtCore.QSize(400, 300))
        Form.setMaximumSize(QtCore.QSize(400, 300))
        self.textBrowser = QtWidgets.QTextBrowser(Form)
        self.textBrowser.setGeometry(QtCore.QRect(0, 0, 400, 300))
        self.textBrowser.setObjectName("textBrowser")

        self.retranslateUi(Form)
        QtCore.QMetaObject.connectSlotsByName(Form)

    def retranslateUi(self, Form):
        Form.setWindowTitle(QtWidgets.QApplication.translate("Form", "Help", None))
        self.textBrowser.setHtml(QtWidgets.QApplication.translate("Form", "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.0//EN\" \"http://www.w3.org/TR/REC-html40/strict.dtd\">\n"
"<html><head><meta name=\"qrichtext\" content=\"1\" /><style type=\"text/css\">\n"
"p, li { white-space: pre-wrap; }\n"
"</style></head><body style=\" font-family:\'Sans\'; font-size:9pt; font-weight:400; font-style:normal;\">\n"
"<p style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">Using Brightness Controller is easy. Most of the parts are self explanatory. Only the advanced parts are discussed here.</p>\n"
"<p style=\"-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><br /></p>\n"
"<p style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><span style=\" font-weight:600;\">Reverse Control</span></p>\n"
"<p style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">Check it only when your primary display is being detected as secondary display. This reverses the control of your monitor. Only available when multiple displays are connected.</p>\n"
"<p style=\"-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><br /></p>\n"
"<p style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><span style=\" font-weight:600;\">Color Temperature</span></p>\n"
"<p style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">Assigns color temperature, also known as color tones to display.</p>\n"
"<p style=\"-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><br /></p>\n"
"<p style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><span style=\" font-weight:600;\">Save Settings</span></p>\n"
"<p style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">Saves your current color configurations</p>\n"
"<p style=\"-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><br /></p>\n"
"<p style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><span style=\" font-weight:600;\">Load Settings</span></p>\n"
"<p style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">Loads your saved color configuration.</p></body></html>"))



src/ui/about.py
# -*- coding: utf-8 -*-

from PyQt5 import QtCore, QtWidgets

class Ui_Form(object):
    def setupUi(self, Form):
        Form.setObjectName("Form")
        Form.resize(400, 300)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Fixed, QtWidgets.QSizePolicy.Fixed)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(Form.sizePolicy().hasHeightForWidth())
        Form.setSizePolicy(sizePolicy)
        Form.setMinimumSize(QtCore.QSize(400, 300))
        Form.setMaximumSize(QtCore.QSize(400, 300))
        self.textBrowser = QtWidgets.QTextBrowser(Form)
        self.textBrowser.setGeometry(QtCore.QRect(0, 0, 401, 301))
        self.textBrowser.setObjectName("textBrowser")

        self.retranslateUi(Form)
        QtCore.QMetaObject.connectSlotsByName(Form)

    def retranslateUi(self, Form):
        Form.setWindowTitle(QtWidgets.QApplication.translate("Form", "About", None))
        self.textBrowser.setHtml(QtWidgets.QApplication.translate("Form", "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.0//EN\" \"http://www.w3.org/TR/REC-html40/strict.dtd\">\n"
"<html><head><meta name=\"qrichtext\" content=\"1\" /><style type=\"text/css\">\n"
"p, li { white-space: pre-wrap; }\n"
"</style></head><body style=\" font-family:\'Sans\'; font-size:9pt; font-weight:400; font-style:normal;\">\n"
"<p style=\" margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><span style=\" font-size:16pt;\">Brightness Controller v2.0 build 1</span></p>\n"
"<p style=\" margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><span style=\" font-weight:600;\">Author: </span>Amit Seal Ami, </p>\n"
"<p style=\" margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">Bangladesh, </p>\n"
"<p style=\" margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">amitseal@outlook.com</p>\n"
"<p style=\" margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><span style=\" font-weight:600;\">Contributors</span></p>\n"
"<p style=\" margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">Significant contributors are listed here based on the contributions.</p>\n"
"<p style=\" margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">- Archisman Panigrahi</p>\n"
"<p style=\" margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">- Zlatan Vasović</p>\n"
"<p style=\" margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">For full list of authors and details, please check https://github.com/lordamit/Brightness</p></body></html>", None))



src/ui/license.py
# -*- coding: utf-8 -*-

from PyQt5 import QtCore, QtWidgets

class Ui_Form(object):
    def setupUi(self, Form):
        Form.setObjectName("Form")
        Form.resize(480, 290)
        Form.setMinimumSize(QtCore.QSize(480, 290))
        Form.setMaximumSize(QtCore.QSize(480, 290))
        self.textBrowser = QtWidgets.QTextBrowser(Form)
        self.textBrowser.setGeometry(QtCore.QRect(0, 0, 481, 290))
        self.textBrowser.setObjectName("textBrowser")

        self.retranslateUi(Form)
        QtCore.QMetaObject.connectSlotsByName(Form)

    def retranslateUi(self, Form):
        Form.setWindowTitle(QtWidgets.QApplication.translate("Form", "License", None))
        self.textBrowser.setHtml(QtWidgets.QApplication.translate("Form", "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.0//EN\" \"http://www.w3.org/TR/REC-html40/strict.dtd\">\n"
"<html><head><meta name=\"qrichtext\" content=\"1\" /><style type=\"text/css\">\n"
"p, li { white-space: pre-wrap; }\n"
"</style></head><body style=\" font-family:\'Sans\'; font-size:9pt; font-weight:400; font-style:normal;\">\n"
"<p align=\"justify\" style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><span style=\" font-weight:600;\">Brightness Controller</span></p>\n"
"<p align=\"justify\" style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">Linux Application to control brightness of dual monitor.</p>\n"
"<p align=\"justify\" style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">Copyright (C) 2013  Amit Seal Ami</p>\n"
"<p align=\"justify\" style=\"-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><br /></p>\n"
"<p align=\"justify\" style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">This program is free software: you can redistribute it and/or modify</p>\n"
"<p align=\"justify\" style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">it under the terms of the GNU General Public License as published by</p>\n"
"<p align=\"justify\" style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">the Free Software Foundation, either version 3 of the License, or</p>\n"
"<p align=\"justify\" style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">(at your option) any later version.</p>\n"
"<p align=\"justify\" style=\"-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><br /></p>\n"
"<p align=\"justify\" style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">This program is distributed in the hope that it will be useful,</p>\n"
"<p align=\"justify\" style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">but WITHOUT ANY WARRANTY; without even the implied warranty of</p>\n"
"<p align=\"justify\" style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the</p>\n"
"<p align=\"justify\" style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">GNU General Public License for more details.</p>\n"
"<p align=\"justify\" style=\"-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><br /></p>\n"
"<p align=\"justify\" style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">You should have received a copy of the GNU General Public License</p>\n"
"<p align=\"justify\" style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">along with this program.  If not, see http://www.gnu.org/licenses/gpl.html.</p>\n"
"<p align=\"justify\" style=\"-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><br /></p>\n"
"<p align=\"justify\" style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">contact: amitseal@outlook.com / lordamit@gmail.com</p></body></html>", None))



src/util/read_config.py
#!/usr/bin/env python

# This file is part of Brightness Controller.
#
# Brightness Controller is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# Brightness Controller is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with Brightness Controller.  If not, see <http://www.gnu.org/licenses/>.

import configparser


def read_configuration(file_path):
    """
    reads configuration from given file path
    For two displays:
    return (p_brightness, p_red, p_green, p_blue,
                       s_brightness, s_red, s_green, s_blue, s_reversed)
    for one display:
    return (p_brightness, p_red, p_green, p_blue)
    """
    config = configparser.ConfigParser()
    config.read(file_path)
    p_brightness = config.getint('primary', 'brightness')
    p_red = config.getint('primary', 'red')
    p_green = config.getint('primary', 'green')
    p_blue = config.getint('primary', 'blue')

    if config.getboolean('primary', 'has_secondary'):
        p_source = config.get('primary', 'source')
        s_brightness = config.getint('secondary', 'brightness')
        s_red = config.getint('secondary', 'red')
        s_green = config.getint('secondary', 'green')
        s_blue = config.getint('secondary', 'blue')
        s_source = config.get('secondary', 'source')
        return (p_brightness, p_red, p_green, p_blue, p_source,
                s_brightness, s_red, s_green, s_blue, s_source)
    else:
        return p_brightness, p_red, p_green, p_blue
        


src/util/write_config.py
#!/usr/bin/env python

# This file is part of Brightness Controller.
#
# Brightness Controller is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# Brightness Controller is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with Brightness Controller.  If not, see <http://www.gnu.org/licenses/>.

import configparser


def write_primary_display(p_br_rgb, file_path):
    """
    writes the configuration file as set in brightness controller
    p_br_rgb - (int primary_brightness, int primary_red,
    int primary_green, int primary_blue)
    @rtype : object
    """
    config = configparser.RawConfigParser()
    config.add_section('type')
    config.add_section('primary')
    config.set('primary', 'has_secondary', False)
    if p_br_rgb is None:
        config.set('primary', 'brightness', 99)
        config.set('primary', 'red', 99)
        config.set('primary', 'green', 99)
        config.set('primary', 'blue', 99)
    else:
        config.set('primary', 'brightness', p_br_rgb[0])
        config.set('primary', 'red', p_br_rgb[1])
        config.set('primary', 'green', p_br_rgb[2])
        config.set('primary', 'blue', p_br_rgb[3])

    with open(file_path, 'w') as configfile:
        config.write(configfile)


def write_both_display(p_br_rgb, s_br_rgb, file_path):
    """
    writes the configuration file as set in brightness controller
    `p_br_rgb` - (int primary_brightness, int primary_red,
    int primary_green, int primary_blue, source)
    s_br_rgb - (int secondary_brightness, int secondary_red,
    int secondary_green, int secondary_blue, source)
    file_path - the save file path
    """
    config = configparser.RawConfigParser()
    config.add_section('primary')
    config.set('primary', 'has_secondary', True)
    if p_br_rgb is None:
        config.set('primary', 'brightness', 99)
        config.set('primary', 'red', 99)
        config.set('primary', 'green', 99)
        config.set('primary', 'blue', 99)
    else:
        config.set('primary', 'brightness', p_br_rgb[0])
        config.set('primary', 'red', p_br_rgb[1])
        config.set('primary', 'green', p_br_rgb[2])
        config.set('primary', 'blue', p_br_rgb[3])
        config.set('primary', 'source', p_br_rgb[4])
    config.add_section('secondary')
    if s_br_rgb is None:
        config.set('secondary', 'brightness', 99)
        config.set('secondary', 'red', 99)
        config.set('secondary', 'green', 99)
        config.set('secondary', 'blue', 99)
    else:
        config.set('secondary', 'brightness', s_br_rgb[0])
        config.set('secondary', 'red', s_br_rgb[1])
        config.set('secondary', 'green', s_br_rgb[2])
        config.set('secondary', 'blue', s_br_rgb[3])
        config.set('secondary', 'source', s_br_rgb[4])
    with open(file_path, 'w') as configfile:
        config.write(configfile)


src/ui/mainwindow.py
# -*- coding: utf-8 -*-

from PyQt5 import QtCore, QtGui
from PyQt5 import QtWidgets

class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(701, 333)
        MainWindow.setMinimumSize(QtCore.QSize(701, 333))
        MainWindow.setMaximumSize(QtCore.QSize(701, 333))
        icon = QtGui.QIcon()
        icon.addPixmap(QtGui.QPixmap("/usr/share/icons/hicolor/scalable/apps/brightness-controller.svg"), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        MainWindow.setWindowIcon(icon)
        self.centralWidget = QtWidgets.QWidget(MainWindow)
        self.centralWidget.setObjectName("centralWidget")
        self.horizontalLayoutWidget = QtWidgets.QWidget(self.centralWidget)
        self.horizontalLayoutWidget.setGeometry(QtCore.QRect(10, 0, 511, 261))
        self.horizontalLayoutWidget.setObjectName("horizontalLayoutWidget")
        self.horizontalLayout = QtWidgets.QHBoxLayout(self.horizontalLayoutWidget)
        self.horizontalLayout.setContentsMargins(0, 0, 0, 0)
        self.horizontalLayout.setObjectName("horizontalLayout")
        self.groupBox_2 = QtWidgets.QGroupBox(self.horizontalLayoutWidget)
        self.groupBox_2.setObjectName("groupBox_2")
        self.horizontalLayoutWidget_2 = QtWidgets.QWidget(self.groupBox_2)
        self.horizontalLayoutWidget_2.setGeometry(QtCore.QRect(10, 40, 231, 211))
        self.horizontalLayoutWidget_2.setObjectName("horizontalLayoutWidget_2")
        self.horizontalLayout_2 = QtWidgets.QHBoxLayout(self.horizontalLayoutWidget_2)
        self.horizontalLayout_2.setContentsMargins(0, 0, 0, 0)
        self.horizontalLayout_2.setObjectName("horizontalLayout_2")
        self.primary_brightness = QtWidgets.QSlider(self.horizontalLayoutWidget_2)
        self.primary_brightness.setProperty("value", 99)
        self.primary_brightness.setOrientation(QtCore.Qt.Vertical)
        self.primary_brightness.setObjectName("primary_brightness")
        self.horizontalLayout_2.addWidget(self.primary_brightness)
        self.primary_red = QtWidgets.QSlider(self.horizontalLayoutWidget_2)
        self.primary_red.setProperty("value", 99)
        self.primary_red.setOrientation(QtCore.Qt.Vertical)
        self.primary_red.setObjectName("primary_red")
        self.horizontalLayout_2.addWidget(self.primary_red)
        self.primary_green = QtWidgets.QSlider(self.horizontalLayoutWidget_2)
        self.primary_green.setProperty("value", 99)
        self.primary_green.setOrientation(QtCore.Qt.Vertical)
        self.primary_green.setObjectName("primary_green")
        self.horizontalLayout_2.addWidget(self.primary_green)
        self.primary_blue = QtWidgets.QSlider(self.horizontalLayoutWidget_2)
        self.primary_blue.setProperty("value", 99)
        self.primary_blue.setOrientation(QtCore.Qt.Vertical)
        self.primary_blue.setObjectName("primary_blue")
        self.horizontalLayout_2.addWidget(self.primary_blue)
        self.horizontalLayout.addWidget(self.groupBox_2)
        self.groupBox = QtWidgets.QGroupBox(self.horizontalLayoutWidget)
        self.groupBox.setObjectName("groupBox")
        self.horizontalLayoutWidget_3 = QtWidgets.QWidget(self.groupBox)
        self.horizontalLayoutWidget_3.setGeometry(QtCore.QRect(20, 40, 231, 211))
        self.horizontalLayoutWidget_3.setObjectName("horizontalLayoutWidget_3")
        self.horizontalLayout_3 = QtWidgets.QHBoxLayout(self.horizontalLayoutWidget_3)
        self.horizontalLayout_3.setContentsMargins(0, 0, 0, 0)
        self.horizontalLayout_3.setObjectName("horizontalLayout_3")
        self.secondary_brightness = QtWidgets.QSlider(self.horizontalLayoutWidget_3)
        self.secondary_brightness.setProperty("value", 99)
        self.secondary_brightness.setOrientation(QtCore.Qt.Vertical)
        self.secondary_brightness.setObjectName("secondary_brightness")
        self.horizontalLayout_3.addWidget(self.secondary_brightness)
        self.secondary_red = QtWidgets.QSlider(self.horizontalLayoutWidget_3)
        self.secondary_red.setProperty("value", 99)
        self.secondary_red.setOrientation(QtCore.Qt.Vertical)
        self.secondary_red.setObjectName("secondary_red")
        self.horizontalLayout_3.addWidget(self.secondary_red)
        self.secondary_green = QtWidgets.QSlider(self.horizontalLayoutWidget_3)
        self.secondary_green.setProperty("value", 99)
        self.secondary_green.setOrientation(QtCore.Qt.Vertical)
        self.secondary_green.setObjectName("secondary_green")
        self.horizontalLayout_3.addWidget(self.secondary_green)
        self.secondary_blue = QtWidgets.QSlider(self.horizontalLayoutWidget_3)
        self.secondary_blue.setProperty("value", 99)
        self.secondary_blue.setOrientation(QtCore.Qt.Vertical)
        self.secondary_blue.setObjectName("secondary_blue")
        self.horizontalLayout_3.addWidget(self.secondary_blue)
        self.horizontalLayout.addWidget(self.groupBox)
        self.label = QtWidgets.QLabel(self.centralWidget)
        self.label.setGeometry(QtCore.QRect(20, 260, 71, 16))
        self.label.setObjectName("label")
        self.label_2 = QtWidgets.QLabel(self.centralWidget)
        self.label_2.setGeometry(QtCore.QRect(100, 260, 21, 16))
        self.label_2.setObjectName("label_2")
        self.label_3 = QtWidgets.QLabel(self.centralWidget)
        self.label_3.setGeometry(QtCore.QRect(150, 260, 21, 16))
        self.label_3.setObjectName("label_3")
        self.label_4 = QtWidgets.QLabel(self.centralWidget)
        self.label_4.setGeometry(QtCore.QRect(200, 260, 16, 16))
        self.label_4.setObjectName("label_4")
        self.label_5 = QtWidgets.QLabel(self.centralWidget)
        self.label_5.setGeometry(QtCore.QRect(290, 260, 71, 16))
        self.label_5.setObjectName("label_5")
        self.label_6 = QtWidgets.QLabel(self.centralWidget)
        self.label_6.setGeometry(QtCore.QRect(370, 260, 21, 16))
        self.label_6.setObjectName("label_6")
        self.label_7 = QtWidgets.QLabel(self.centralWidget)
        self.label_7.setGeometry(QtCore.QRect(420, 260, 21, 16))
        self.label_7.setObjectName("label_7")
        self.label_8 = QtWidgets.QLabel(self.centralWidget)
        self.label_8.setGeometry(QtCore.QRect(470, 260, 16, 16))
        self.label_8.setObjectName("label_8")
        self.verticalLayoutWidget = QtWidgets.QWidget(self.centralWidget)
        self.verticalLayoutWidget.setGeometry(QtCore.QRect(529, 0, 213, 204))
        self.verticalLayoutWidget.setObjectName("verticalLayoutWidget")
        self.verticalLayout = QtWidgets.QVBoxLayout(self.verticalLayoutWidget)
        self.verticalLayout.setSizeConstraint(QtWidgets.QLayout.SetDefaultConstraint)
        self.verticalLayout.setContentsMargins(0, -1, 0, -1)
        self.verticalLayout.setObjectName("verticalLayout")
        self.label_9 = QtWidgets.QLabel(self.verticalLayoutWidget)
        self.label_9.setObjectName("label_9")
        self.verticalLayout.addWidget(self.label_9)
        self.label_12 = QtWidgets.QLabel(self.verticalLayoutWidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Preferred, QtWidgets.QSizePolicy.Preferred)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.label_12.sizePolicy().hasHeightForWidth())
        self.label_12.setSizePolicy(sizePolicy)
        self.label_12.setObjectName("label_12")
        self.verticalLayout.addWidget(self.label_12)
        self.primary_combobox = QtWidgets.QComboBox(self.verticalLayoutWidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.primary_combobox.sizePolicy().hasHeightForWidth())
        self.primary_combobox.setSizePolicy(sizePolicy)
        self.primary_combobox.setMaximumSize(QtCore.QSize(150, 28))
        self.primary_combobox.setObjectName("primary_combobox")
        self.verticalLayout.addWidget(self.primary_combobox)
        self.label_11 = QtWidgets.QLabel(self.verticalLayoutWidget)
        self.label_11.setObjectName("label_11")
        self.verticalLayout.addWidget(self.label_11)
        self.secondary_combo = QtWidgets.QComboBox(self.verticalLayoutWidget)
        self.secondary_combo.setMaximumSize(QtCore.QSize(150, 28))
        self.secondary_combo.setObjectName("secondary_combo")
        self.verticalLayout.addWidget(self.secondary_combo)
        self.label_10 = QtWidgets.QLabel(self.verticalLayoutWidget)
        self.label_10.setObjectName("label_10")
        self.verticalLayout.addWidget(self.label_10)
        self.comboBox = QtWidgets.QComboBox(self.verticalLayoutWidget)
        self.comboBox.setMaximumSize(QtCore.QSize(150, 28))
        self.comboBox.setObjectName("comboBox")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.comboBox.addItem("")
        self.verticalLayout.addWidget(self.comboBox)
        MainWindow.setCentralWidget(self.centralWidget)
        self.statusBar = QtWidgets.QStatusBar(MainWindow)
        self.statusBar.setObjectName("statusBar")
        MainWindow.setStatusBar(self.statusBar)
        self.menuBar = QtWidgets.QMenuBar(MainWindow)
        self.menuBar.setGeometry(QtCore.QRect(0, 0, 701, 26))
        self.menuBar.setObjectName("menuBar")
        self.menuFile = QtWidgets.QMenu(self.menuBar)
        self.menuFile.setObjectName("menuFile")
        self.menuHelp = QtWidgets.QMenu(self.menuBar)
        self.menuHelp.setObjectName("menuHelp")
        MainWindow.setMenuBar(self.menuBar)
        self.actionExit = QtWidgets.QAction(MainWindow)
        self.actionExit.setObjectName("actionExit")
        self.actionHelp = QtWidgets.QAction(MainWindow)
        self.actionHelp.setObjectName("actionHelp")
        self.actionAbout = QtWidgets.QAction(MainWindow)
        self.actionAbout.setObjectName("actionAbout")
        self.actionLicense = QtWidgets.QAction(MainWindow)
        self.actionLicense.setObjectName("actionLicense")
        self.actionSave = QtWidgets.QAction(MainWindow)
        self.actionSave.setObjectName("actionSave")
        self.actionLoad = QtWidgets.QAction(MainWindow)
        self.actionLoad.setObjectName("actionLoad")
        self.actionCheck_for_update = QtWidgets.QAction(MainWindow)
        self.actionCheck_for_update.setEnabled(False)
        self.actionCheck_for_update.setObjectName("actionCheck_for_update")
        self.menuFile.addAction(self.actionSave)
        self.menuFile.addAction(self.actionLoad)
        self.menuFile.addSeparator()
        self.menuFile.addAction(self.actionExit)
        self.menuHelp.addAction(self.actionHelp)
        self.menuHelp.addAction(self.actionAbout)
        self.menuHelp.addAction(self.actionLicense)
        self.menuHelp.addAction(self.actionCheck_for_update)
        self.menuBar.addAction(self.menuFile.menuAction())
        self.menuBar.addAction(self.menuHelp.menuAction())

        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        MainWindow.setWindowTitle(QtWidgets.QApplication.translate("MainWindow", "Brightness Controller", None))
        self.groupBox_2.setTitle(QtWidgets.QApplication.translate("MainWindow", "Primary", None))
        self.groupBox.setTitle(QtWidgets.QApplication.translate("MainWindow", "Secondary", None))
        self.label.setText(QtWidgets.QApplication.translate("MainWindow", "Brightness", None))
        self.label_2.setText(QtWidgets.QApplication.translate("MainWindow", " R", None))
        self.label_3.setText(QtWidgets.QApplication.translate("MainWindow", " G", None))
        self.label_4.setText(QtWidgets.QApplication.translate("MainWindow", " B", None))
        self.label_5.setText(QtWidgets.QApplication.translate("MainWindow", "Brightness", None))
        self.label_6.setText(QtWidgets.QApplication.translate("MainWindow", " R", None))
        self.label_7.setText(QtWidgets.QApplication.translate("MainWindow", " G", None))
        self.label_8.setText(QtWidgets.QApplication.translate("MainWindow", " B", None))
        self.label_9.setText(QtWidgets.QApplication.translate("MainWindow", "Advanced", None))
        self.label_12.setText(QtWidgets.QApplication.translate("MainWindow", "Primary Brightness", None))
        self.label_11.setText(QtWidgets.QApplication.translate("MainWindow", "Secondary Brightness", None))
        self.secondary_combo.setToolTip(QtWidgets.QApplication.translate("MainWindow", "<html><head/><body><p>You can select the source of Secondary Brightness Here.</p></body></html>", None))
        self.label_10.setText(QtWidgets.QApplication.translate("MainWindow", "Color Temperature", None))
        self.comboBox.setItemText(0, QtWidgets.QApplication.translate("MainWindow", "Default", None))
        self.comboBox.setItemText(1, QtWidgets.QApplication.translate("MainWindow", "1900K Candle", None))
        self.comboBox.setItemText(2, QtWidgets.QApplication.translate("MainWindow", "2600K 40W Tungsten", None))
        self.comboBox.setItemText(3, QtWidgets.QApplication.translate("MainWindow", "2850K 100W Tungsten", None))
        self.comboBox.setItemText(4, QtWidgets.QApplication.translate("MainWindow", "3200K Halogen", None))
        self.comboBox.setItemText(5, QtWidgets.QApplication.translate("MainWindow", "5200K Carbon Arc", None))
        self.comboBox.setItemText(6, QtWidgets.QApplication.translate("MainWindow", "5400K High Noon", None))
        self.comboBox.setItemText(7, QtWidgets.QApplication.translate("MainWindow", "6000K Direct Sun", None))
        self.comboBox.setItemText(8, QtWidgets.QApplication.translate("MainWindow", "7000K Overcast Sky", None))
        self.comboBox.setItemText(9, QtWidgets.QApplication.translate("MainWindow", "20000K Clear Blue Sky", None))
        self.menuFile.setTitle(QtWidgets.QApplication.translate("MainWindow", "&File", None))
        self.menuHelp.setTitle(QtWidgets.QApplication.translate("MainWindow", "&Help", None))
        self.actionExit.setText(QtWidgets.QApplication.translate("MainWindow", "E&xit", None))
        self.actionHelp.setText(QtWidgets.QApplication.translate("MainWindow", "&Help", None))
        self.actionAbout.setText(QtWidgets.QApplication.translate("MainWindow", "&About", None))
        self.actionLicense.setText(QtWidgets.QApplication.translate("MainWindow", "&License", None))
        self.actionSave.setText(QtWidgets.QApplication.translate("MainWindow", "&Save current settings", None))
        self.actionLoad.setText(QtWidgets.QApplication.translate("MainWindow", "&Load settings", None))
        self.actionCheck_for_update.setText(QtWidgets.QApplication.translate("MainWindow", "&Check for Update", None))

